### PR TITLE
14 Bürokratt processes Participants' requests to start using Bürokratt

### LINKED ISSUE #14: Bürokratt processes Participants' requests to start using Bürokratt
AS A Bürokratt Project manager
I WANT TO process Participants' requests to start using Bürokratt
SO THAT I would have that process automated and easy as much as possbile

#### General requirements

- [ ] List all REST endpoints created and used to fulfill the AC of this issue as valid OpenAPI specs
- [ ] User-provided input must be validated against pre-approved options defined in an appropriate Ruuter DSL

#### Endpoints

- [ ] #72 to process Participants' request to start using Bürokratt

#### Business requirements

- [ ] Only users with appropriate permissions have access to process Participants' requests by re-using the template of #23 
- [ ] GUI is used to process Participants' requests

#### Expected functionalities

> All of the functionalities below are based on Postgres' database table

- [ ] List all requests to start using Bürokratt
- [ ] Update the state of the application according to https://github.com/buerokratt/Buerokratt-Architectural-Decision-Records/issues/32
- [ ] Update the content of the Participant's information
- [ ] Mark request as deleted

#### DSL

- [ ] Use `institutions` database table based on #24

#### GUI

- [ ] Follows the principles of https://github.com/e-gov/cvi and https://e-gov.github.io/cvi/?path=/story/intro--page
- [ ] Uses https://github.com/buerokratt/cvi as technical stack _(PS! Currently in progress to get React-based components there)_
- [ ] Input for GUI fields are fetched from an appropriate configuration file

### PR DESCRIPTION
**Pull request by karljogi**
Issue #14 

Adds an institutions table so the foreign keys could exist for participants.

To test this, at least 1 institution has to be entered to the database. 

Many changes stem from the latest technical setup update: #89

### CODE REVIEW

File: DSL/Ruuter.private/DELETE/admin/participants.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# check_for_cookie:
+#   switch:
+#     - condition: ${incoming.headers == null || incoming.headers.cookie == null}
+#       next: return_unauthorized
+
+authenticate:
+  template: templates/permissions/admin
+  requestType: post
+  headers:
+    cookie: cookie #${incoming.headers.cookie}
```
Comment (1AhmedYasser) [inline DSL/Ruuter.private/DELETE/admin/participants.yml:11]: Note: To be replaced with tim jwt cookie (In Ruuter Guards)

File: DSL/Ruuter.private/GET/admin/institutions.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# check_for_cookie:
+#   switch:
+#     - condition: ${incoming.headers == null || incoming.headers.cookie == null}
+#       next: return_unauthorized
+
+authenticate:
+  template: templates/permissions/admin
+  requestType: post
+  headers:
+    cookie: cookie #${incoming.headers.cookie}
```
Comment (1AhmedYasser) [inline DSL/Ruuter.private/GET/admin/institutions.yml:11]: Note: To be replaced with tim jwt cookie (In Ruuter Guards)

File: DSL/Ruuter.private/GET/admin/participants/applications.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# check_for_cookie:
+#   switch:
+#     - condition: ${incoming.headers == null || incoming.headers.cookie == null}
+#       next: return_unauthorized
+
+authenticate:
+  template: templates/permissions/admin
+  requestType: post
+  headers:
+    cookie: cookie #${incoming.headers.cookie}
```
Comment (1AhmedYasser) [inline DSL/Ruuter.private/GET/admin/participants/applications.yml:11]: To be replaced with tim jwt cookie (In Ruuter Guards)

File: DSL/Ruuter.private/PUT/admin/participants.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+# check_for_cookie:
+#   switch:
+#     - condition: ${incoming.headers == null || incoming.headers.cookie == null}
+#       next: return_unauthorized
+
+authenticate:
+  template: templates/permissions/admin
+  requestType: post
+  headers:
+    cookie: cookie #${incoming.headers.cookie}
```
Comment (1AhmedYasser) [inline DSL/Ruuter.private/PUT/admin/participants.yml:11]: Note: To be replaced with tim jwt cookie (In Ruuter Guards)

File: DSL/Ruuter.private/PUT/admin/participants/status.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+# check_for_cookie:
+#   switch:
+#     - condition: ${incoming.headers == null || incoming.headers.cookie == null}
+#       next: return_unauthorized
+
+authenticate:
+  template: templates/permissions/admin
+  requestType: post
+  headers:
+    cookie: cookie #${incoming.headers.cookie}
```
Comment (1AhmedYasser) [inline DSL/Ruuter.private/PUT/admin/participants/status.yml:11]: Note: To be replaced with tim jwt cookie (In Ruuter Guards)

File: DSL/Ruuter.public/POST/templates/permissions/applicant.yml [hunk 6]
```diff
@@ -2,7 +2,7 @@ extract_token_data:
   call: http.post
   args:
     # TODO: Use real TIM endpoint
```
Comment (1AhmedYasser) [inline DSL/Ruuter.public/POST/templates/permissions/applicant.yml:4]: Note: To be replaced with tim jwt cookie (In Ruuter Guards)

File: README.md [hunk 7]
```diff
@@ -20,6 +20,8 @@ Bürokratt Central Operations
 
 - If you made changes to data mapper, rebuild the image again using `docker build -t docker build -t datamapper .`
 
+- Ruuter configuration should be changed for all the endpoints to work. PUT and DELETE should be added to the allowedMethodTypes array. See more information https://github.com/buerokratt/Ruuter/blob/65889552329249665e48656ed866cdf99cda391f/samples/CONFIGURATION.md
```
Comment (1AhmedYasser) [inline README.md:23]: Linked to 

- https://github.com/buerokratt/Ruuter/issues/164
- https://github.com/buerokratt/Ruuter/issues/165

**Review by 1AhmedYasser — APPROVED** [review]

**Review by baha-a — APPROVED** [review]

File: DSL/Ruuter.private/DELETE/admin/participants.yml (hunk 1 above)
Comment (turnerrainer) [reply DSL/Ruuter.private/DELETE/admin/participants.yml:11]: Ok

File: DSL/Ruuter.private/GET/admin/institutions.yml (hunk 2 above)
Comment (turnerrainer) [reply DSL/Ruuter.private/GET/admin/institutions.yml:11]: Ok

File: DSL/Ruuter.private/GET/admin/participants/applications.yml (hunk 3 above)
Comment (turnerrainer) [reply DSL/Ruuter.private/GET/admin/participants/applications.yml:11]: Ok

File: DSL/Ruuter.private/PUT/admin/participants.yml (hunk 4 above)
Comment (turnerrainer) [reply DSL/Ruuter.private/PUT/admin/participants.yml:11]: Ok

File: DSL/Ruuter.private/PUT/admin/participants/status.yml (hunk 5 above)
Comment (turnerrainer) [reply DSL/Ruuter.private/PUT/admin/participants/status.yml:11]: Ok

File: DSL/Ruuter.public/POST/templates/permissions/applicant.yml (hunk 6 above)
Comment (turnerrainer) [reply DSL/Ruuter.public/POST/templates/permissions/applicant.yml:4]: Ok

**Review by turnerrainer — APPROVED** [review]
Approve as reviewed by @1AhmedYasser and @baha-a

**Pull request merged**